packages/backend/src/redirect.ts

```diff
@@ -0,0 +1,67 @@
+import { parsePublishableKey } from './util/parsePublishableKey';
+
+type RedirectAdapter = (url: string) => any;
+
+type SingUpParams = { url?: string; returnBackUrl?: string };
+type SingInParams = { url?: string; returnBackUrl?: string };
+
+const buildPath = (destinationUrl: string, currentUrl?: string) => {
+  let destinationURL;
+  if (!destinationUrl.startsWith('http')) {
+    if (!currentUrl || !currentUrl.startsWith('http')) {
+      throw new Error('destination url or return back url should be an absolute path url!');
+    }
+
+    const currentURL = new URL(currentUrl);
+    destinationURL = new URL(destinationUrl, currentURL.origin);
+  } else {
+    destinationURL = new URL(destinationUrl);
+  }
+
+  if (currentUrl) {
+    destinationURL.searchParams.set('redirect_url', currentUrl);
+  }
+
+  return destinationURL.toString();
+};
+
+type RedirectParams = {
+  redirectAdapter: RedirectAdapter;
+  signInUrl?: string;
+  signUpUrl?: string;
+  publishableKey?: string;
+  frontendApi?: string;
+};
+
+export function redirect({ redirectAdapter, signUpUrl, signInUrl, frontendApi, publishableKey }: RedirectParams) {
+  const accountsBaseUrl = getAccountsBaseUrl(frontendApi, publishableKey);
+
+  const redirectToSignUp = ({ returnBackUrl }: SingUpParams = {}) => {
+    if (!signUpUrl && !accountsBaseUrl) {
+      throw new Error('signUpUrl or frontendApi/publishableKey should be provided!');
```

Comment:
❓ Do we have a format for errors thrown from @clerk/backend?

Reply:
as far as know, we don't:
- `packages/backend/src/util/assertValidSecretKey.ts#L4`
- `packages/backend/src/tokens/request.ts#L89`